### DISCUSSION TITLE
How to customize error message when a command fails?

### DISCUSSION THREAD

**Discussion by Swandog** [casey/just#1721] [General]
I've got a recipe like this:

```just
check:
    cargo fmt --all --check
    cargo check --workspace
    cargo clippy --workspace
    cargo deny check
```

When this fails, the message I get is simply
`error: Recipe `check` failed on line 38 with exit code 1`
which works, but doesn't tell me what command actually failed. So I always have to look it up and then figure out how it was run (because my memory is bad). I'd like to, instead, be able to print out the command that failed, or an error message telling me what to run to diagnose it.

What's the best way to output a better message? I can do it with some bash scripting but I'm wondering if there's a simpler way.

**Comment by casey** [comment]
Just should echo each command before running it, so you should see the command that failed right before the failure. Does that help?

**Reply by Swandog** [reply]
Yeah, it does that; but when you have a very noisy command like `clippy`, that gets lost in the terminal.

I know it's a bit minor but I'd really just like something that saves me from having to scroll up to find the failing command. This is not crucial, just something that would make my life easier

**Reply by casey** [reply]
Makes sense, I'll open an issue.
